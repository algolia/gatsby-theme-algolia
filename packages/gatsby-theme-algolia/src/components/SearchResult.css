.ais-SearchBox-input::-ms-clear,
.ais-SearchBox-input::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.ais-SearchBox-input::-webkit-search-decoration,
.ais-SearchBox-input::-webkit-search-cancel-button,
.ais-SearchBox-input::-webkit-search-results-button,
.ais-SearchBox-input::-webkit-search-results-decoration {
  display: none;
}

.ais-SearchBox-form {
  display: block;
  position: relative;
}

.ais-SearchBox-input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0.3rem 1.7rem;
  width: 100%;
  position: relative;
  background-color: #fff;
  border: 1px solid #c4c8d8;
  border-radius: 5px;
}
.ais-SearchBox-input::-webkit-input-placeholder {
  color: grey;
}
.ais-SearchBox-input::-moz-placeholder {
  color: grey;
}
.ais-SearchBox-input:-ms-input-placeholder {
  color: grey;
}
.ais-SearchBox-input:-moz-placeholder {
  color: grey;
}

.ais-SearchBox-submit,
.ais-SearchBox-reset,
.ais-SearchBox-loadingIndicator {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: absolute;
  z-index: 1;
  width: 20px;
  height: 20px;
  top: 50%;
  right: 0.3rem;
  -webkit-transform: translateY(-50%);
  transform: translateY(-50%);
}

.ais-SearchBox-submit {
  left: 0.3rem;
}

.ais-SearchBox-reset {
  right: 0.3rem;
}

.ais-SearchBox-submitIcon,
.ais-SearchBox-resetIcon,
.ais-SearchBox-loadingIcon {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.ais-SearchBox-submitIcon path,
.ais-SearchBox-resetIcon path {
  fill: grey;
}

.ais-SearchBox-submitIcon {
  width: 14px;
  height: 14px;
}

.ais-SearchBox-resetIcon {
  width: 12px;
  height: 12px;
}

.ais-SearchBox-loadingIcon {
  width: 16px;
  height: 16px;
}

.ais-Highlight-highlighted {
  background: orange;
}
