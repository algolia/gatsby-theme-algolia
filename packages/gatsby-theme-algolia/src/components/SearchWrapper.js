import React from 'react';
import algoliasearch from 'algoliasearch/lite';
import { InstantSearch } from 'react-instantsearch-dom';

import 'instantsearch.css/themes/reset.css';

const checkRequiredValues = (
  algoliaAppId,
  algoliaSearchApiKey,
  algoliaIndexName
) => {
  [
    { value: algoliaAppId, name: 'algoliaAppId' },
    { value: algoliaSearchApiKey, name: 'algoliaSearchApiKey' },
    { value: algoliaIndexName, name: 'algoliaIndexName' },
  ].forEach(({ value, name }) => {
    if (!value) {
      throw new Error(`${name} is missing!`);
    }
  });
};

export default ({
  algoliaAppId = process.env.GATSBY_ALGOLIA_APP_ID,
  algoliaSearchApiKey = process.env.GATSBY_ALGOLIA_SEARCH_API_KEY,
  algoliaIndexName = process.env.GATSBY_ALGOLIA_INDEX_NAME,
  children,
}) => {
  checkRequiredValues(algoliaAppId, algoliaSearchApiKey, algoliaIndexName);

  const searchClient = algoliasearch(algoliaAppId, algoliaSearchApiKey);
  return (
    <InstantSearch indexName={algoliaIndexName} searchClient={searchClient}>
      {children}
    </InstantSearch>
  );
};
