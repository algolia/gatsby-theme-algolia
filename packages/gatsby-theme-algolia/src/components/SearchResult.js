import React from 'react';
import { Link } from 'gatsby';
import { Hits, Highlight } from 'react-instantsearch-dom';

import './SearchResult.css';

const defaultHitComponent = ({ hit }) => {
  const { path, date } = hit;
  return (
    <div key={path}>
      <h3>
        <Link to={path}>
          <Highlight hit={hit} attribute="title" />
        </Link>
      </h3>
      <small>{new Date(date).toDateString()}</small>
      <p>
        <Highlight hit={hit} attribute="description" />
      </p>
    </div>
  );
};

export default ({ hitComponent = defaultHitComponent, ...otherProps }) => (
  <Hits hitComponent={hitComponent} {...otherProps} />
);
