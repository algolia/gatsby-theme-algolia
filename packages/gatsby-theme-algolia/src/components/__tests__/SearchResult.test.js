import React from 'react';
import SearchWrapper from '../SearchWrapper';
import SearchResult from '../SearchResult';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from '@testing-library/react';

describe('Rendering', () => {
  it('renders without arguments', () => {
    const { container } = render(
      <SearchWrapper
        algoliaAppId="a"
        algoliaSearchApiKey="b"
        algoliaIndexName="c"
      >
        <SearchResult />
      </SearchWrapper>
    );
    expect(container).toMatchSnapshot();
  });
});
