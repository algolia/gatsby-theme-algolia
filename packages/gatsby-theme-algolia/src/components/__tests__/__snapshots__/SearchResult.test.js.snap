// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders without arguments 1`] = `
<div>
  <div
    class="ais-InstantSearch__root"
  >
    <div
      class="ais-Hits"
    >
      <ul
        class="ais-Hits-list"
      />
    </div>
  </div>
</div>
`;
